test/functional/DDR_Test/src/j9vm/test/corehelper/CoreGen.java

```diff
@@ -26,7 +26,22 @@
 
  	public static void main(String[] args) {
 		SimpleThread st = new SimpleThread();
+
+		/* Fork a thread for TestJITExt that is guaranteed to have a JIT frame
+		 * at the top of the stack. Fix until  https://github.com/eclipse/openj9/issues/5966 
+		 * is resolved.
+		 */
+		Compiler.compileClass(j9vm.test.corehelper.TestJITExtHelperThread.class);
+		TestJITExtHelperThread tjet = new TestJITExtHelperThread();
+
+		tjet.configureJittedHelperThread();
+
+		/* wait to start core dump until thread is in a waiting state */
+		while(!tjet.isThreadWaiting()) {}
```

Comment:
This should really be protected by the same synchronization as the other users of the field, i.e. wait until the forked thread indicates it's waiting. I think the code might be best in the `configureJittedHelperThread`function, so when that returns, the forked thread is in the desired state.